h/api/nipsa/search.py

```diff
@@ -0,0 +1,61 @@
+def nipsa_filter(userid=None):
+    """Return an Elasticsearch filter for filtering out NIPSA'd annotations.
+
+    The returned filter is suitable for inserting into an Es query dict.
+    For example:
+
+        query = {
+            "query": {
+                "filtered": {
+                    "filter": nipsa_filter(),
+                    "query": {...}
+                }
+            }
+        }
+
+    :param userid: The ID of a user whose annotations should not be filtered.
+        The returned filtered query won't filter out this user's annotations,
+        even if the annotations have the NIPSA flag.
+    :type userid: unicode
+
+    """
+    # If any one of these "should" clauses is true then the annotation will
+    # get through the filter.
+    should_clauses = [{"not": {"term": {"nipsa": True}}}]
+
+    if userid:
```

Comment:
I don't really understand. If you pass `user=123`, or `user=sys.stdout`, this code is still going to do The Wrong Thing. As far as I'm concerned, this is again a sentinel-vs-falsy distinction: the API of this function says "the `userid` parameter must be a valid userid, or the sentinel None, to indicate that there is no authenticated user". So we we should be precise and check for that sentinel.